src/mqtt_client.rs

```diff
@@ -73,7 +73,11 @@ where
             Some(index) => self.session_state.pending_subscriptions.swap_remove(index),
         };
 
-        for code in subscribe_acknowledge.codes.iter() {
+        for code in subscribe_acknowledge
+            .codes
+            .iter()
+            .map(|&x| ReasonCode::from(x))
+        {
             code.as_result()?;
         }
```

Comment:
Merging the `map` and loop body code:

```suggestion
        for &code in subscribe_acknowledge.codes.iter() {
            ReasonCode::from(x).as_result()?;
        }
```